### ISSUE TITLE
epic: Gas Price Equilibrium

### ISSUE THREAD

**Issue by eugypalu** [kkrt-labs/kakarot#1526]
### Description

In the current design, we:

- relay transactions for users (`execute_from_outside` feature)
    - relayers pay the Starknet fee associated with their Kakarot transaction
- charge users in Kakarot Core EVM wrt to consumed EVM gas and signed gas price (base fee and max priority fee)

### Problem

We need to find a way to balance relayers operational costs and user EVM fees

Non-exhaustive ideas and options:

1. In legacy transactions, there is only one concept of fee: gas price. We could use the gas price parameter to make sure the user pays (user fee := gasUsed* gasPrice) for the Starknet costs. Note that we know in advance the expected Starknet cost as well as the EVM gas a specific transaction consumes.
TL;DR: Force legacy transactions on Kakarot & set `gasPrice := starknet_total_estimated_fee/EVM_gas`
2. We are able to observe dynamically Kakarot transactions on Starknet public and compute the average relayer cost per transaction as well as the average EVM gas used. And then, dynamically set the baseFee of Kakarot: `baseFee := average_relayer_cost_per_transaction/average_EVM_gas`

- [ ] #1328 
- [ ] relayers-observer that checks the relayers' balance every X minutes and funds them if necessary
- [ ] fee-balancer that checks the fee dynamics every X minutes and rebalances the fees
- [ ] Set up CloudWatch alarms

**Comment by CanutoIX** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.
